common/Utils.java

```diff
@@ -73,7 +73,7 @@ public static String readNonEmptyLine(LineReader reader, String prompt) throws I
     }
 
     public static String readPassword(LineReader passwordReader, String prompt) {
-        return passwordReader.readLine(prompt, (char) 0);
+        return passwordReader.readLine(prompt, '*');
```

Comment:
(char) 0 would make the prompt empty, making it unclear whether a password was being typed at all. Now, this asterisk is given in place. This does leak the length of the password being typed in exchange for some UX.

Reply:
I agree that the UX would be better this way.